applications/OptimizationApplication/python_scripts/controls/shape/vertex_morphing_shape_control.py

```diff
@@ -0,0 +1,369 @@
+        self.SetupMeshMotion()
+
+        self.is_initialized = False
+
+    @time_decorator(methodName="GetName")
+    def Initialize(self) -> None:
+
+        if self.filter_type == "bulk_surface_implicit":
+            self.filter_model_part = self.model_part_operation.GetModelPart().GetRootModelPart()
+        else:
+            self.filter_model_part = self.model_part_operation.GetModelPart()
+
+        # initialize filters
+        if not self.is_filter_implicit:
+            # make sure that explicit filter model part only has conditions
+            if len(self.filter_model_part.Elements) > 0:
+                raise RuntimeError("VertexMorphingShapeControl with explicit filtering only allows model parts with conditions")
+            # now damping and creating the filter
+            fixed_model_parts_names = list(self.parameters["fixed_model_parts"].keys())
+            if len(fixed_model_parts_names)>0:
+                self.fixed_model_part_operation = ModelPartOperation(self.model, ModelPartOperation.OperationType.UNION, f"control_{self.GetName()}", fixed_model_parts_names, False)
+                self.fixed_model_part = self.fixed_model_part_operation.GetModelPart()
+                self.filter = KratosOA.NodalExplicitFilter(self.filter_model_part, self.fixed_model_part, self.parameters["filter_settings"]["filter_function_type"].GetString(),
+                                                           self.parameters["filter_settings"]["damping_function_type"].GetString(),
+                                                           self.parameters["filter_settings"]["max_nodes_in_filter_radius"].GetInt())
+            else:
+                self.filter = KratosOA.NodalExplicitFilter(self.filter_model_part, self.parameters["filter_settings"]["filter_function_type"].GetString(),
+                                                           self.parameters["filter_settings"]["max_nodes_in_filter_radius"].GetInt())
+
+            filter_radius_field = Kratos.Expression.NodalExpression(self.filter_model_part)
+            Kratos.Expression.LiteralExpressionIO.SetData(filter_radius_field, self.parameters["filter_settings"]["radius"].GetDouble())
+            self.filter.SetFilterRadius(filter_radius_field)
+            self.control_field = Kratos.Expression.NodalExpression(self.filter_model_part)
+            Kratos.Expression.LiteralExpressionIO.SetData(self.control_field,[0,0,0])
+        else:
+            self.filter.Initialize()
+            self._SetFixedModelPartValues(False)
+            self.control_field = self.filter.UnFilterField(self.GetPhysicalField())
+            self._SetFixedModelPartValues(True)
+
+        # initialize_mesh_motion
+        if self.has_mesh_motion:
+            self._mesh_moving_analysis.Initialize()
+
+        # now update
+        self._UpdateAndOutputFields(self.GetEmptyField())
+        self.is_initialized = True
+
+    def SetupMeshMotion(self):
+        self.has_mesh_motion = True
```

Comment:
initialize all self variables in the constructor as None or settings value

Reply:
fixed. Let's merge it then

Reply:
SetupMeshMotion is called in init so now should be fine